crates/types/src/nao_image.rs

```diff
@@ -0,0 +1,245 @@
+use std::{
+    fmt::Debug,
+    mem::{size_of, ManuallyDrop},
+    ops::Index,
+    path::Path,
+    sync::Arc,
+};
+
+use color_eyre::eyre::{self, WrapErr};
+use image::{
+    codecs::jpeg::JpegEncoder, io::Reader, load_from_memory_with_format, ImageError, ImageFormat,
+    RgbImage,
+};
+use nalgebra::Point2;
+use serde::{Deserialize, Serialize};
+use serialize_hierarchy::{DecodeJpeg, EncodeJpeg, SerializeHierarchy};
+
+use crate::{Rgb, YCbCr422, YCbCr444};
+
+#[derive(Clone, Default, Serialize, Deserialize, SerializeHierarchy, Debug)]
+#[serialize_hierarchy(as_jpeg)]
+pub struct NaoImage {
```

Comment:
I disagree. This image is not a `YCbCrImage`

Reply:
I thought, we already decided that `NaoImage` does not play well together with `GrayscaleImage`. `YCbCrImage` would be my proposal, because then the image prefixes tell which color space is represented. Feel free to propose a different naming rule, but `Nao` does not fit when applying the color space rule.

Reply:
`YCbCr422Image` or `Yuv422Image` ?

Reply:
Yes, both sound better than `NaoImage`. Feel free to decide and go ahead.

Reply:
Done